File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shaders/uniforms/Vec4iUniform.java
package org.saar.lwjgl.opengl.shaders.uniforms;

import org.joml.Vector4ic;
import org.lwjgl.opengl.GL20;

public abstract class Vec4iUniform extends UniformBase implements Uniform {

    @Override
    public final void doLoad() {
        final Vector4ic v = getUniformValue();
        GL20.glUniform4i(getLocation(), v.x(), v.y(), v.z(), v.w());
    }

    public abstract Vector4ic getUniformValue();
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/shader/uniforms/Vec4iUniform.kt
package org.saar.lwjgl.opengl.shader.uniforms

import org.joml.Vector4ic
import org.lwjgl.opengl.GL20

abstract class Vec4iUniform : Uniform {

    final override fun load(location: Int) {
        val value: Vector4ic = this.value
        GL20.glUniform4i(location, value.x(), value.y(), value.z(), value.w())
    }

    abstract val value: Vector4ic
}